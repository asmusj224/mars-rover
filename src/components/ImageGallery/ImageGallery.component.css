.image-gallery {
  max-width: 1200px;
  margin: 0 auto;
  margin-top: 2rem; }
  .image-gallery__list {
    list-style: none;
    display: flex;
    flex-wrap: wrap; }
  .image-gallery__item {
    display: flex;
    flex-direction: column;
    width: 250px;
    margin: 0 1rem 1rem 1rem;
    transition: all 0.3s; }
    .image-gallery__item:hover {
      transform: scale(1.1); }
  .image-gallery__image {
    max-width: 100%;
    display: block;
    opacity: 0.7;
    height: 250px;
    border-radius: 3px; }
    .image-gallery__image:hover {
      opacity: 1; }
  .image-gallery__description {
    text-align: center;
    font-size: 14px;
    color: #a9a9a9; }
