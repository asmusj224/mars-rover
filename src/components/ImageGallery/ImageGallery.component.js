import React from 'react';
import './ImageGallery.component.css';

const ImageGallery = ({ roverImages = [] }) => {
    roverImages = roverImages.map(roverImage => {
        return (<li className="image-gallery__item" key={roverImage.id}>
            <img className="image-gallery__image" src={roverImage.img_src} alt={roverImage.rover.name} />
            <span className="image-gallery__description">{roverImage.rover.name} - {roverImage.camera.full_name}</span>
        </li>)
    });


    return (
        <div className="image-gallery">
            <ul className="image-gallery__list">{roverImages}</ul>
        </div>
    );
}

export default ImageGallery;