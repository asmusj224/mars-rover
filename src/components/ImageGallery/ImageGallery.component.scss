@import 'styles/_colors.scss';
.image-gallery {
    max-width: 1200px;
    margin: 0 auto;
    margin-top: 2rem;
    &__list {
        list-style: none;
        display: flex;
        flex-wrap: wrap;
    }
    &__item {
        display: flex;
        flex-direction: column;
        width: 250px;
        margin: 0 1rem 1rem 1rem;
        transition: all 0.3s;
        &:hover {
            transform: scale(1.1);
        }
    }
    &__image {
        max-width: 100%;
        display: block;
        opacity: 0.7;
        height: 250px;
        border-radius: 3px;
        &:hover {
            opacity: 1;
        }
    }
    &__description {
        text-align: center;
        font-size: 14px;
        color: $darkgray;
    }
}