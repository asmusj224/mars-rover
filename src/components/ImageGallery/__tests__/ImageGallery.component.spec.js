import React from 'react';
import renderer from 'react-test-renderer';

import ImageGallery from '../ImageGallery.component';

const roverImages = [{
    "id": 634105,
    "sol": 1847,
    "camera": {
        "id": 20,
        "name": "FHAZ",
        "rover_id": 5,
        "full_name": "Front Hazard Avoidance Camera"
    },
    "img_src": "http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01847/opgs/edr/fcam/FLB_561457773EDR_F0661516FHAZ00341M_.JPG",
    "earth_date": "2017-10-16",
    "rover": {
        "id": 5,
        "name": "Curiosity",
        "landing_date": "2012-08-06",
        "launch_date": "2011-11-26",
        "status": "active",
        "max_sol": 2134,
        "max_date": "2018-08-07",
        "total_photos": 340100,
        "cameras": [
            {
                "name": "FHAZ",
                "full_name": "Front Hazard Avoidance Camera"
            },
            {
                "name": "NAVCAM",
                "full_name": "Navigation Camera"
            },
            {
                "name": "MAST",
                "full_name": "Mast Camera"
            },
            {
                "name": "CHEMCAM",
                "full_name": "Chemistry and Camera Complex"
            },
            {
                "name": "MAHLI",
                "full_name": "Mars Hand Lens Imager"
            },
            {
                "name": "MARDI",
                "full_name": "Mars Descent Imager"
            },
            {
                "name": "RHAZ",
                "full_name": "Rear Hazard Avoidance Camera"
            }
        ]
    }
}]

describe('<ImageGallery />', () => {
    it('renders correctly', () => {
        const component = renderer.create(
            <ImageGallery roverImages={roverImages} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});