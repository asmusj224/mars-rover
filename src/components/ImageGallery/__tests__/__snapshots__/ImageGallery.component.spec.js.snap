// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageGallery /> renders correctly 1`] = `
<div
  className="image-gallery"
>
  <ul
    className="image-gallery__list"
  >
    <li
      className="image-gallery__item"
    >
      <img
        alt="Curiosity"
        className="image-gallery__image"
        src="http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01847/opgs/edr/fcam/FLB_561457773EDR_F0661516FHAZ00341M_.JPG"
      />
      <span
        className="image-gallery__description"
      >
        Curiosity
         - 
        Front Hazard Avoidance Camera
      </span>
    </li>
  </ul>
</div>
`;
